### DISCUSSION TITLE
[Linux] Run with local ollama

### DISCUSSION THREAD

**Discussion by omnigenous** [TheR1D/shell_gpt#467] [Q&A] [answered]
I was following a wiki: https://github.com/TheR1D/shell_gpt/wiki/Ollama

But could get it to work with shell_gpt, works fine with `ollama run mistral:7b-instruct`

Running `sgpt --model ollama/mistral:7b-instruct  "Who are you?"` gives error:
```
Usage: sgpt [OPTIONS] [PROMPT]
Try 'sgpt --help' for help.
╭─ Error ─────────────────────────────────────────────────────────────────────────────────────────────╮
│ Missing config key: OPENAI_API_KEY                                                                  │
╰─────────────────────────────────────────────────────────────────────────────────────────────────────╯
```

I tried changing `DEFAULT_MODEL` to `ollama/mistral:7b-instruct`, `OPENAI_BASE_URL` to `localhost`, `OPENAI_USE_FUNCTIONS` to `false` and `OPENAI_API_KEY` to `false` in `~/.config/shell_gpt/.sgptrc`. Latest change gives long error message so I revert it.

I also saw: https://github.com/TheR1D/shell_gpt/discussions/336#discussioncomment-8350429

Tried:
```
git clone https://github.com/TheR1D/shell_gpt.git
cd shell_gpt
git checkout ollama
python -m venv venv
source venv/bin/activate
pip install -e  .

sgpt --model ollama/mistral:7b-instruct --no-functions "Hi Ollama"
```

Same "Missing config key: OPENAI_API_KEY" error

**Comment by TheR1D** [comment]
Hi @omnigenous make sure you have `OPENAI_API_KEY` set to any value in `~/.config/shell_gpt/.sgptrc`.

**Comment by omnigenous** [comment, marked as answer]
Installing `shell-gpt` version `1.3.0-1` from aur worked.

And like @TheR1D said in https://github.com/TheR1D/shell_gpt/discussions/467#discussioncomment-8435278, don't forget to set `OPENAI_API_KEY` to any value like `false` in `~/.config/shell_gpt/.sgptrc`

Remember to run local ollama `ollama serve` in background ;)

**Comment by gwpl** [comment]
Seems like new issue arised: https://github.com/TheR1D/shell_gpt/discussions/503
